p2p/memp2p/memp2p_test.go

```diff
@@ -0,0 +1,220 @@
+package memp2p_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/memp2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitializingNetworkwith4Peers(t *testing.T) {
```

Comment:
This test could be split into multiple other tests.
One test for creating a network
one for creating a messenger, etc.
Then there is no need for so many asserts in one test.

Reply:
Rewritten and split into smaller, more focused tests.